sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -653,6 +653,167 @@ class SparkSqlAstBuilder extends AstBuilder {
     })
   }
 
+  /**
+   * Create a [[CreateUserDefinedFunctionCommand]].
+   *
+   * For example:
+   * {{{
+   *   CREATE [OR REPLACE] [TEMPORARY] FUNCTION [IF NOT EXISTS] [db_name.]function_name
+   *   ([param_name param_type [COMMENT param_comment], ...])
+   *   RETURNS {ret_type | TABLE (ret_name ret_type [COMMENT ret_comment], ...])}
+   *   [routine_characteristics]
+   *   RETURN {expression | TABLE ( query )};
```

Comment:
I think the PR description is copied from here, but we need to update here to match the full syntax. @allisonwang-db

Reply:
The `SECURITY-` clause and its KEYWORDS are not necessary to be introduced based on the privilege control of other objects like tables, columns, and other host language UDFs.

Reply:
Yes, and Spark fails if the SECURITY clause is specified: https://github.com/apache/spark/pull/46816/files#diff-77a9aad2da3dc60210a2c4d2f3165d5f1d0acd54ca4811072a053225170ed748R808

It's just for better error message: instead of antlr errors, we give a clear error message for unsupported features.

Reply:
Having better error messages on unsupported features sounds reasonable to me. However, we seems to ban a MYSQL feature instead of the ANSI one.
- ANSI 1999
`<external security clause> ::= EXTERNAL SECURITY DEFINER | EXTERNAL SECURITY INVOKER | EXTERNAL SECURITY IMPLEMENTATION DEFINED`
- MySQL:
https://dev.mysql.com/doc/refman/8.4/en/stored-objects-security.html

Reply:
Maybe higher ANSI standards support these, but I don't have a copy to check:)

Reply:
cc @srielau for ANSI SQL syntax.

Reply:
Good catch. I've updated the description for this PR and will create a follow up PR to fix it in code.